sdk_v2/kubeflow/training/utils/utils.py

```diff
@@ -166,8 +166,8 @@ def get_args_using_train_func(
                 read -r -d '' SCRIPT << EOM\n
                 {func_code}
                 EOM
-                printf "%s" \"$SCRIPT\" > \"{func_file}\"
-                {entrypoint} \"{func_file}\""""
+                printf "%s" \"$SCRIPT\" > \"$program_path\"/\"{func_file}\"
+                {entrypoint} \"$program_path\"/\"{func_file}\""""
```

Comment:
If we create this file under `HOME` directory, will it still fail for non-root containers ?

I intentionally write the script into the same file, so the file system between users' workspace and Kubernetes Pod is the same. Which means users can use standard Python imports while developing ML training code. Similar to how we showcase at KubeCon demo: https://youtu.be/Lgy4ir1AhYw?t=525

Related: https://github.com/kubeflow/training-operator/issues/2347.

cc @shravan-achar

Reply:
With the stricter security policy, containers are run with random UID with no fixed `HOME` directory.

I need to research a bit more what could be the other possible options and get back to you.

Reply:
I've given this another cycle and it doesn't seem possible to change the working directory permissions easily.

On the other hand, mounting an `emptyDir` volume as the `/workspace` directory in the training runtime works as expected with the stricter security policy.

My understanding is this approach aligns well with the initializer and worker pods coordinating via PVCs. The default `emptyDir` fits with the initializers being currently run as init containers, but this can be adapted to mount a PVC when they'll run as separate Jobs.